ToolBox/CoinApi/CoinApiDataQueueHandler.cs

```diff
@@ -326,5 +341,113 @@ private void OnError(object sender, Exception e)
         {
             Log.Error(e);
         }
+
+        #region SynchronizingHistoryProvider
+
+        public override void Initialize(HistoryProviderInitializeParameters parameters)
+        {
+            // NOP
+        }
+
+        public override IEnumerable<Slice> GetHistory(IEnumerable<HistoryRequest> requests, DateTimeZone sliceTimeZone)
+        {
+            var subscriptions = new List<Subscription>();
+            foreach (var request in requests)
+            {
+                var history = GetHistory(request);
+                var subscription = CreateSubscription(request, history);
+                subscriptions.Add(subscription);
+            }
+            return CreateSliceEnumerableFromSubscriptions(subscriptions, sliceTimeZone);
+        }
+
+        public IEnumerable<BaseData> GetHistory(HistoryRequest historyRequest)
+        {
+            if (historyRequest.Symbol.SecurityType != SecurityType.Crypto)
+            {
+                Log.Error($"CoinApiDataQueueHandler.GetHistory(): Invalid security type {historyRequest.Symbol.SecurityType}");
+                yield break;
+            }
+
+            if (historyRequest.Resolution == Resolution.Tick)
+            {
+                Log.Error("CoinApiDataQueueHandler.GetHistory(): No historical ticks, only OHLCV timeseries");
+                yield break;
+            }
+
+            if (historyRequest.DataType == typeof(QuoteBar))
+            {
+                Log.Error("CoinApiDataQueueHandler.GetHistory(): No historical QuoteBars , only TradeBars");
+                yield break;
+            }
+
+            // Perform a check the number of bars requested, this must not exceed the limit of 100k
+            var dataRequestedCount = (historyRequest.EndTimeUtc - historyRequest.StartTimeUtc).Ticks 
+                                     / historyRequest.Resolution.ToTimeSpan().Ticks;
+
+            if (dataRequestedCount > 100000)
+            {
+                Log.Error("CoinApiDataQueueHandler.GetHistory(): The requested data range is too large, " +
```

Comment:
Think the history request could be paginated and performed in more than one call?

Reply:
Hi @Martin-Molinero thanks for review. I will fix this on Monday, sorry little bit busy week for me here..

Reply:
Thanks! 👍